### ISSUE TITLE
Planning: calculate supply-driven & demand-driven post-save

### ISSUE THREAD

**Issue by fosterlynn** [Carbon-Farm-Network/app-carbon-farm-network#69]
1 week

This need some discussion with the users, including if to do it,  how to do it, and the priority.

**Comment by fosterlynn** [comment]
From testing myself, thinking about needs during their season, and then discussion with the users:  We will want a version of this during the season, as actual events are entered.  At each stage of the supply chain, they will want to re-schedule forward based on what actually happened at the last completed stage.

This should be done based on the relative quantities in the plan, not based on the recipe.

I think we could start with a schedule forward one stage at a time, rather than the whole plan.  Or both.  But one stage at a time is not that hard.

There are details to be worked out:
* should the option be by column (process specification) or by input or output column?
* should it only be allowed starting at the last column with events entered?  If not, what are the rules where events have been entered?
* probably several others......

**Comment by fosterlynn** [comment]
This issue includes both supply-driven and demand-driven recalculations now.  They can probably be done in the same manner, with consistent UX.  More design needed.

**Comment by fosterlynn** [comment]
This is looking really great!  :heart: 

One thing that isn't the way I understood we talked about,  let's discuss if this isn't just a bug.  When going forward from the output of Ship to Spin, and looking at the inputs to Spin, they have gone out of balance with the recipe.  I.e. the multiplying out of the recipe quantities didn't stop when the first input ran out of room.

(Lower priority than #148 .)

**Comment by fosterlynn** [comment]
Some time, let's make the icon for this more prominent, matching the other icons.  Maybe just an arrow if that is easier?

**Comment by LeosPrograms** [comment]
> When going forward from the output of Ship to Spin, and looking at the inputs to Spin, they have gone out of balance with the recipe. I.e. the multiplying out of the recipe quantities didn't stop when the first input ran out of room.

Yes, this was a choice because it is unclear which inputs are required for a recipe to be valid. Many seem to be optional. Sometimes there are inputs that are required, but don't need to be proportional (e.g. gas and truck rental).

**Comment by fosterlynn** [comment]
>Yes, this was a choice because it is unclear which inputs are required for a recipe to be valid.

OK.  I'll check with the users when we have it running and they can see the implications.  I wonder if there's a way to tell the difference, like if it is a "one" unit?  Anyhow, let's leave it, I'll ask.

**Comment by fosterlynn** [comment]
Closing!  Will add demand-driven and bugs/enhancements as they come up as separate issues.
